test/types/populate.test.ts

```diff
@@ -149,4 +149,34 @@ function gh11321(): void {
       return 'foo';
     }
   });
+}
+
+function gh11496() {
+  interface Friend {
+    blocked: boolean
+  }
+  const FriendSchema = new Schema<Friend>({
+    blocked: Boolean
+  });
+
+  interface User {
+    friends: Types.ObjectId[];
+  }
+  const UserSchema = new Schema<User>({
+    friends: [{ type: Schema.Types.ObjectId, ref: 'friends' }]
+  });
+  const Users = model<User>('friends', UserSchema);
+
+
+  Users.findOne({}).populate('friends').then(user => {
+    expectType<Types.ObjectId | undefined>(user?.friends[0]);
+    expectError(user?.friends[0].blocked);
+    expectError(user?.friends.map(friend => friend.blocked));
+  });
+
+  Users.findOne({}).populate<{friends: Friend[]}>('friends').then(user => {
+    expectAssignable<Friend | undefined>(user?.friends[0]);
+    user?.friends[0].blocked;
```

Comment:
Can we do `expectType<boolean>` here?

Reply:
I've tried to write two tests for this & it doesn't success, although it shows correct intellisens even in map function.

![Untitled](https://user-images.githubusercontent.com/50488054/157557133-7444d37f-b7d4-4372-9c8c-0a42330fbd79.png)

Reply:
So is it now fixed?